### DISCUSSION TITLE
Is there an issue with the list function..?

### DISCUSSION THREAD

**Discussion by fwd-maker** [anasty17/mirror-leech-telegram-bot#1217] [Q&A]
I experienced a strange thing. That is, the list function sometimes doesn't respond when I use the function. Strangely this happened when I rarely used the function, and it took me to use the list function (command) twice. I've included the logs below this text

Beginning of logs:
app_1  | 2022-11-16 00:58:22,611 - bot - INFO - listing: miui
app_1  | 2022-11-16 00:58:22,612 - bot.helper.mirror_utils.upload_utils.gdriveTools - INFO - Authorizing with 98.json service account
app_1  | Exception in thread Thread-5:
app_1  | Traceback (most recent call last):
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 703, in urlopen
app_1  |     httplib_response = self._make_request(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 449, in _make_request
app_1  |     six.raise_from(e, None)
app_1  |   File "<string>", line 3, in raise_from
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 444, in _make_request
app_1  |     httplib_response = conn.getresponse()
app_1  |   File "/usr/lib/python3.8/http/client.py", line 1348, in getresponse
app_1  |     response.begin()
app_1  |   File "/usr/lib/python3.8/http/client.py", line 316, in begin
app_1  |     version, status, reason = self._read_status()
app_1  |   File "/usr/lib/python3.8/http/client.py", line 277, in _read_status
app_1  |     line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
app_1  |   File "/usr/lib/python3.8/socket.py", line 669, in readinto
app_1  |     return self._sock.recv_into(b)
app_1  |   File "/usr/lib/python3.8/ssl.py", line 1241, in recv_into
app_1  |     return self.read(nbytes, buffer)
app_1  |   File "/usr/lib/python3.8/ssl.py", line 1099, in read
app_1  |     return self._sslobj.read(len, buffer)
app_1  | ConnectionResetError: [Errno 104] Connection reset by peer
app_1  | 
app_1  | During handling of the above exception, another exception occurred:
app_1  | 
app_1  | Traceback (most recent call last):
app_1  |   File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 489, in send
app_1  |     resp = conn.urlopen(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 787, in urlopen
app_1  |     retries = retries.increment(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py", line 550, in increment
app_1  |     raise six.reraise(type(error), error, _stacktrace)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/packages/six.py", line 769, in reraise
app_1  |     raise value.with_traceback(tb)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 703, in urlopen
app_1  |     httplib_response = self._make_request(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 449, in _make_request
app_1  |     six.raise_from(e, None)
app_1  |   File "<string>", line 3, in raise_from
app_1  |   File "/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py", line 444, in _make_request
app_1  |     httplib_response = conn.getresponse()
app_1  |   File "/usr/lib/python3.8/http/client.py", line 1348, in getresponse
app_1  |     response.begin()
app_1  |   File "/usr/lib/python3.8/http/client.py", line 316, in begin
app_1  |     version, status, reason = self._read_status()
app_1  |   File "/usr/lib/python3.8/http/client.py", line 277, in _read_status
app_1  |     line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
app_1  |   File "/usr/lib/python3.8/socket.py", line 669, in readinto
app_1  |     return self._sock.recv_into(b)
app_1  |   File "/usr/lib/python3.8/ssl.py", line 1241, in recv_into
app_1  |     return self.read(nbytes, buffer)
app_1  |   File "/usr/lib/python3.8/ssl.py", line 1099, in read
app_1  |     return self._sslobj.read(len, buffer)
app_1  | urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
app_1  | 
app_1  | During handling of the above exception, another exception occurred:
app_1  | 
app_1  | Traceback (most recent call last):
app_1  |   File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
app_1  |     self.run()
app_1  |   File "/usr/lib/python3.8/threading.py", line 870, in run
app_1  |     self._target(*self._args, **self._kwargs)
app_1  |   File "/usr/src/app/bot/modules/list.py", line 43, in _list_drive
app_1  |     msg, button = gdrive.drive_list(key, isRecursive=True, itemType=item_type)
app_1  |   File "/usr/src/app/bot/helper/mirror_utils/upload_utils/gdriveTools.py", line 607, in drive_list
app_1  |     telegraph.create_page(
app_1  |   File "/usr/src/app/bot/helper/ext_utils/telegraph_helper.py", line 30, in create_page
app_1  |     return self.telegraph.create_page(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/telegraph/api.py", line 183, in create_page
app_1  |     return self._telegraph.method('createPage', values={
app_1  |   File "/usr/local/lib/python3.8/dist-packages/telegraph/api.py", line 32, in method
app_1  |     response = self.session.post(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 635, in post
app_1  |     return self.request("POST", url, data=data, json=json, **kwargs)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 587, in request
app_1  |     resp = self.send(prep, **send_kwargs)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/requests/sessions.py", line 701, in send
app_1  |     r = adapter.send(request, **kwargs)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/requests/adapters.py", line 547, in send
app_1  |     raise ConnectionError(err, request=request)
app_1  | requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
app_1  | 2022-11-16 00:59:11,408 - bot - INFO - listing: miui
app_1  | 2022-11-16 00:59:11,408 - bot.helper.mirror_utils.upload_utils.gdriveTools - INFO - Authorizing with 34.json service account

**Comment by anasty17** [comment]
Connection issues

**Reply by fwd-maker** [reply]
okey thank u, maybe issue on my server connection

**Reply by anasty17** [reply]
/eval from requests import get
r = get("https://graph.org/")
print(r.status_code)

Try this in ur bot and reply the result here.

**Reply by fwd-maker** [reply]
OK, here's a bit of a weird thing I found. Namely this problem occurs only when I use GCP and I didn't find this problem when I use AWS, oh yes for the command you sent has the following result:

AWS Server:
> /eval from requests import get
r = get("https://graph.org/")
print(r.status_code)
>200

GCP Server:
> /eval from requests import get
r = get("https://graph.org/")
print(r.status_code)
> 200

For GCP servers, the list function will crash after 20-30 minutes of inactivity

**Reply by anasty17** [reply]
So it's not blocked. If u didn't edit anything in telegraph_helper.py then idk. Response 200 is normal.

**Reply by fwd-maker** [reply]
Yes this is very strange. I didn't edit anything, in fact it looks just by changing the server. And I happen to use the list function very rarely so it's not a big deal, and the function still works when I need it. Just a minor issue,  thankyou for the response

**Reply by anasty17** [reply]
U should know that we are using an alternative domain for telegraph which is [graph](https://graph.org/). This because telegraph blocked in some regions.
